src/scrollSpy/scrollSpyAreaDirective.js

```diff
@@ -0,0 +1,36 @@
+const _window = new WeakMap();
```

Comment:
I think there is no any pluses to hide that fields, especially with weak maps -> it makes code noise and weak maps are not free